dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/AbstractClouderaManagerTask.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager;
+
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.handle.Handle;
+import com.google.edwmigration.dumper.application.dumper.task.AbstractTask;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+public abstract class AbstractClouderaManagerTask extends AbstractTask<Void> {
```

Comment:
I understand that some new classes in this package were designed for inheritance. For each method, I'd recommend to ether document or prohibit overriding it.

For most methods it should be as quick as making them `private` or `final`, but in such cases where subclasses are supposed to override the method, please document the consequences and reasons for overriding.

Reply:
In this class there is `final` method which should not be overridden and the `abstract` which must be overridden.  Additionally, there is a helper method that can be overridden if needed in a subclass (at least I don't expect that right now).  I think it's what you expect. 

Do you expect something different?

Reply:
This `abstract` class marked as package-private. 
If you will find any documents with the motivation to make subclasses private please share and I will just the code